test.py

```diff
@@ -1,47 +1,258 @@
+        )
+
+
+    def move_on_command(self):
+        '''handles movement with keypresses'''
+        ret = [False, False]
+
+        if 'a' in chelone.pressed_keys:
+            ret[0] = True
+            self.move(-7, 0)
+
+            if self.orientation != "left":
+                self.flip()
+
+        elif 'd' in chelone.pressed_keys:
+            ret[1] = True
+            self.move(7, 0)
+
+            if self.orientation != "right":
+                self.flip()
+
+        return ret
+
+    def handle_collision(self, collided_obj, my_collider, other_collider, handled=False):
+        '''check if we ever intersect ground to be able to jump'''
+        displacement = get_collision_displacement(my_collider, other_collider)
+        super().handle_collision(collided_obj, my_collider, other_collider, handled)
+
+        if displacement[1] < 0 and self.vel[1] >= 0:
+            if my_collider.type != "trigger" and other_collider.type != "trigger":
+                self.grounded = True
+
+    def delete_self(self):
+        '''
+        instead of deleting, we need to pause and restart the game
+        TODO: this is a POC, needs to be remade properly
+        '''
+
+        print("game_over")
+        saved_sprites = chelone._sprites
+        chelone._sprites = [{} for i in range(50)]
+        game_over = Sprite(
+            id="game_over", frame=loader.load("game_over.png"),
+            x=450+chelone.camera.x, y=chelone.camera.y, phys_type="immovable", layer=1,
+        )
+        chelone.add_sprite(game_over, 1)
+
+
+        # loop while dead
+        while "Return" not in chelone.pressed_keys:
+            check_keys()
+            chelone.root.update()
+
+        chelone.next_frame = time()
+        chelone.now = time()
+        chelone._sprites = saved_sprites
+
+        super().delete_self()
+
+        player = Player(
+            id="Player", frame=loader.load("tmp.png"),\
+            x=250, y=0, gravity=self.gravity,\
+            state_anim_directory=self.state_anim_directory
+        )
+        chelone.add_sprite(player)
+        game_over.delete_self()
+
+
+class Droid(Sprite):
+    """
+        Basic enemy. Partrols an area and shoots the player if it sees him.
+    """
+
+    class Laser(Sprite):
+        """
+            Projectile that the Droid enemy shoots.
+        """
+        def setup(self, kargs):
+
+            self.velocity = kargs.get("velocity", [-3, 0])
+            self.gravity = 0
+
+
+        def update(self):
+            self.move(self.velocity[0], self.velocity[1])
+
+        def handle_trigger(self, collided_obj, my_collider, other_collider):
+            if not isinstance(collided_obj, Droid):
+                self.delete_self()
+                collided_obj.delete_self()
+                print("Laser attacked " + collided_obj.id)
+
+
+    def setup(self, kargs):
+
+        self.patrol_range = kargs.get("patrol_range", [self.x-300, self.x])
```

Comment:
Sometimes you separate math operators by spaces, sometimes not, we need common style
```suggestion
        self.patrol_range = kargs.get("patrol_range", [self.x - 300, self.x])
```